Plutarch/Lift.hs

```diff
@@ -99,24 +100,27 @@ data LiftError
   = LiftError_EvalError EvalError
   | LiftError_KnownTypeError KnownTypeError
   | LiftError_FromRepr
+  | LiftError_CompilationError Text
   deriving stock (Eq)
 
 {- | Convert a Plutarch term to the associated Haskell value. Fail otherwise.
 This will fully evaluate the arbitrary closed expression, and convert the resulting value.
 -}
-plift' :: forall p. PUnsafeLiftDecl p => ClosedTerm p -> Either LiftError (PLifted p)
-plift' prog = case evalScript (compile prog) of
-  (Right (Scripts.unScript -> UPLC.Program _ _ term), _, _) ->
-    case readKnownConstant term of
-      Right r -> case pconstantFromRepr r of
-        Just h -> Right h
-        Nothing -> Left LiftError_FromRepr
-      Left e -> Left $ LiftError_KnownTypeError e
-  (Left e, _, _) -> Left $ LiftError_EvalError e
+plift' :: forall p. PUnsafeLiftDecl p => Config -> ClosedTerm p -> Either LiftError (PLifted p)
+plift' config prog = case compile config prog of
+  Left msg -> Left $ LiftError_CompilationError msg
+  Right script -> case evalScript script of
```

Comment:
Although not directly relevant, this might be a good time to make this `evalScript` usage here use max budget.